ezgui/src/canvas.rs

```diff
@@ -142,6 +146,24 @@ impl Canvas {
         } else if self.drag_just_ended {
             self.drag_just_ended = false;
         }
+
+        let cursor_screen_pt = self.get_cursor().to_pt();
+        let cursor_map_pt = self.screen_to_map(self.get_cursor());
+        let inner_bounds = self.get_inner_bounds();
+        let map_bounds = self.get_map_bounds();
+        if !inner_bounds.contains(cursor_screen_pt) && !self.disable_edge_panning
+           && map_bounds.contains(cursor_map_pt){
+            let center_pt = self.center_to_screen_pt().to_pt();
+            let displacement_x = cursor_screen_pt.x() - center_pt.x();
+            let displacement_y = cursor_screen_pt.y() - center_pt.y();
+            let displacement_magnitude = f64::sqrt(displacement_x.powf(2.0) + displacement_y.powf(2.0));
+            let displacement_unit_x = displacement_x / displacement_magnitude;
+            let displacement_unit_y = displacement_y / displacement_magnitude;
+            //Add displacement along x axis
```

Comment:
nit: I think one comment could cover both of these, something like "Add displacement along each axis"